packages/libp2p/test/connection-manager/direct.node.ts

```diff
@@ -496,9 +493,11 @@ describe('libp2p.dialer (direct, TCP)', () => {
   })
 
   it('should use the protectors when provided for connecting', async () => {
-    const protector: ConnectionProtector = preSharedKey({
-      psk: swarmKeyBuffer
-    })()
+    const protector: ConnectionProtector = {
+      async protect (connection) {
```

Comment:
since this is just a fake, now, this is a more isolated test to connection-manager's use of protectors, which is good.

do we need to test that the arguments passed to connection protector are correct?

does the protector have tests to confirm it is doing the right thing since we're no longer confirming that as a side-effect of this test?